packages/core/src/system.ts

```diff
@@ -173,11 +207,56 @@ export function createSystem<T extends ActorSystemInfo>(
       if (systemId !== undefined) {
         keyedActors.delete(systemId);
         reverseKeyedActors.delete(actorRef);
+        const event = {
+          type: `@xstate.system.actor.unregister.${systemId as string}`,
+          actorRef: actorRef as T['actors'][keyof T['actors']]
+        } as const;
+        systemSubscriptions.get(systemId)?.forEach((listener) => {
+          listener(event);
+        });
+        systemSubscriptions.get(rootSubscriptionKey)?.forEach((listener) => {
+          listener(event);
+        });
       }
     },
     get: (systemId) => {
       return keyedActors.get(systemId) as T['actors'][any];
     },
+    subscribe: <K extends keyof T['actors']>(
+      maybeSystemIdOrListener:
+        | K
+        | ((event: RegistrationEvent<T['actors'][keyof T['actors']]>) => void),
+      maybeListener?:
+        | ((event: RegistrationEvent<T['actors'][keyof T['actors']]>) => void)
+        | undefined
+    ) => {
+      let listener =
+        typeof maybeSystemIdOrListener === 'function'
+          ? maybeSystemIdOrListener
+          : maybeListener!;
```

Comment:
not sure if this is okay, typescript would prevent this but there's no runtime check